app/components/modals/index.js

```diff
@@ -0,0 +1,15 @@
+import {mapStateToProps} from '/store'
+import Example from './example'
+
+const MODALS = {
+  Example,
+}
+
+export default mapStateToProps(
+  (state) => ({modal: state.modal})
+)(({modal}) => {
+  const name = Object.keys(modal)[0] || ''
+  const Modal = MODALS[name]
```

Comment:
Probably should do some more type checking here, in case the modal comp doesn't exist and either throw, or log a warning.